packages/template-retail-react-app/app/hooks/use-einstein.js

```diff
@@ -56,42 +56,57 @@ export class EinsteinAPI {
      */
     _constructEinsteinProduct(product) {
-        if (product.type && (product.type.master || product.type.variant)) {
-            // handle variants for PDP / viewProduct
-            // Assumes product is a Product object from SCAPI Shopper-Products:
-            // https://developer.salesforce.com/docs/commerce/commerce-api/references/shopper-products?meta=type%3AProduct
+        // Handle variants for PDP / viewProduct
+        // Assumes product is a Product object from SCAPI Shopper-Products:
+        // https://developer.salesforce.com/docs/commerce/commerce-api/references/shopper-products?meta=type%3AProduct
+        if (product.type) {
+            // In case of variant, send the "sku" attribute
+            if (product.type.variant) {
+                return {
+                    id: product.master.masterId,
+                    price: product.price,
+                    sku: product.id
+                }
+            }
+
+            // In case of variation group, send the "altId" and "type" attributes
+            if (product.type.variationGroup) {
+                return {
+                    altId: product.id,
+                    id: product.master.variantId,
+                    price: product.price,
+                    sku: product.id,
+                    type: 'vgroup'
+                }
+            }
+
+            // Handle non-variant products, like master, set, bundle, item.
+            // This code follows the implementation in the plugin_einstein_api.
+            // https://github.com/SalesforceCommerceCloud/plugin_einstein_api/blob/c8168d5b8e2e34bfb9413da73969d59b0f3adabd/plugin_einstein_api/cartridge/scripts/helpers/RecommendationsHelper.js#L315
             return {
-                id: product.master.masterId,
-                sku: product.id,
-                altId: '',
-                altIdType: ''
+                id: product.id,
+                price: product.price
             }
-        } else if (
-            product.productType &&
-            (product.productType.master ||
-                product.productType.variant ||
-                product.productType.set ||
-                product.productType.bundle ||
-                product.productType.variationGroup ||
-                product.productType.item)
-        ) {
-            // handle variants & sets for PLP / viewCategory & viewSearch
-            // Assumes product is a ProductSearchHit from SCAPI Shopper-Search:
-            // https://developer.salesforce.com/docs/commerce/commerce-api/references/shopper-search?meta=type%3AProductSearchHit
+        }
+
+        // Handle variants & sets for PLP / viewCategory & viewSearch
+        // Assumes product is a ProductSearchHit from SCAPI Shopper-Search:
+        // https://developer.salesforce.com/docs/commerce/commerce-api/references/shopper-search?meta=type%3AProductSearchHit
+        if (product.hitType) {
```

Comment:
Nit: we may want to split the handling of products for PDP / PLP into separate functions that are called from `_constructEinsteinProduct`

Reply:
Lets leave that to a follow up PR given the small estimate on this ticket. We are probably over budget as is.